### PR TITLE
fix: Selecting a saved filter in data browser also highlights other filters with equal names

### LINKED ISSUE #2459: Saved filters can incorrectly highlight
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and [Parse Dashboard](https://github.com/parse-community/parse-dashboard/releases). 

### Issue Description

If two saved filters have the same JSON, they can both be highlighted at the same time.

Screenshot 2023-06-11 at 11 29 16 pm

### Steps to reproduce

Create two identical filters across different classes, and select one

### Actual Outcome

Both highlight

### Expected Outcome

Only one should highligh

### Environment

Dashboard
- Parse Dashboard version: `FILL_THIS_OUT`
- Browser (Safari, Chrome, Firefox, Edge, etc.): `FILL_THIS_OUT`
- Browser version: `FILL_THIS_OUT`

Server
- Parse Server version: `FILL_THIS_OUT`
- Operating system: `FILL_THIS_OUT`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `FILL_THIS_OUT`

Database
- System (MongoDB or Postgres): `FILL_THIS_OUT`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

### PR DESCRIPTION
**Pull request by AshishBarvaliya**
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

Closes: #2459 

### Approach

To prevent both different filters from being highlighted, the class names should also be compared.

### CODE REVIEW

File: src/components/CategoryList/CategoryList.react.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ export default class CategoryList extends React.Component {
           let count = c.count;
           let className = id === this.props.current ? styles.active : '';
           let selectedFilter = null;
-          if (this.state.openClasses.includes(id)) {
+            if (this.state.openClasses.includes(id) && id === this.props.current) {
```
Comment (mtrezza) [inline src/components/CategoryList/CategoryList.react.js:105]: It seems we need some linting here

Comment (mtrezza) [inline src/components/CategoryList/CategoryList.react.js]:
```suggestion
          if (this.state.openClasses.includes(id) && id === this.props.current) {
```

**Review by mtrezza — APPROVED** [review]
Looks good, works in preview

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.20](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0-alpha.20)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-beta.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0)

**Pull request merged**